index.js

```diff
@@ -224,8 +339,10 @@ async function createCodeDeployDeployment(codedeploy, clusterName, service, task
     }
   };
   // If it hasn't been set then we don't even want to pass it to the api call to maintain previous behaviour.
+  let codeDeployDescriptionLimit = 512; 
+
   if (codeDeployDescription) {
-    deploymentParams.description = codeDeployDescription
+    deploymentParams.description = (codeDeployDescription.length <= codeDeployDescriptionLimit) ? codeDeployDescription : `${codeDeployDescription.substring(0,511)}…`;
```

Comment:
Since this line of change does not directly relate to the methods implemented in this PR, we might want to exclude this from here. Do correct me if this is infact necessary for this PR?

If we want to make this change, we should create a separate PR, for separation of logic? Let me know what you think: @s3cube @omkhegde